Components/Coroutines/Town/TrinityTownRun.SellItems.cs

```diff
@@ -33,21 +33,21 @@ public static bool ShouldSell(TrinityItem i)
                    !Combat.TrinityCombat.Loot.ShouldStash(i);
         }
 
-        public static async Task<bool> SellItems()
+        public static async Task<CoroutineResult> SellItems()
         {
             if (!ZetaDia.IsInTown)
-                return true;
+                return CoroutineResult.NoAction;
 
             var sellItem = Core.Inventory.Backpack
                 .FirstOrDefault(i => ShouldSell(i) &&
                                      InventoryManager.CanSellItem(i.ToAcdItem()));
             if (sellItem == null)
             {
-                if (!await RepairItems())
-                    return false;
+                if (await RepairItems() == CoroutineResult.Running)
```

Comment:
IMO, return results of coroutines should be handled more robustly, specially for Failure cases. Action to take for Done and NoAction cases seems the same in this call. However, RepairItems() may be returning Failure or may be changed to return Failure for some cases in future and this would break this call without being noticed.  Switching the RepairItems call here would make it a lot more clear and robust.

A failed state from a coroutine should mean that bot will not be able to continue execution properly. And that state should be persisted up to the main entry point if the handler of the call can not make it get rid of a failure condition. For example, while we are trying to vendor items when our inventory is full, failure state from navigation to vendor should be handled by vendoring coroutine to pick a different vendor. If the vendoring coroutine can not succeed even after trying another vendor, it should persist the failure state one level up, so the main coroutine might try reentering the game or such. I am not sure if this was your intention for the Failure state but it makes the most sense to me that way.

There are some other cases like this one. I will not be dropping comments for each those. In general, Failure cases should be handled more robustly.